### ISSUE TITLE
vertical layout

### ISSUE THREAD

**Issue by smaragdus** [tualatin/winmd5checksum#18]
I find the current horizontal layout a bit unhandy- the user needs to use the horizontal scroll-bar too much. I think that it would be easier and handier if the horizontal layout is replaced with vertical layout, for example:

* File 1
  * MD5
  * SHA1
  * SHA256
  * SHA512

* File 2
  * MD5
  * SHA1
  * SHA256
  * SHA512

This way selecting algorithms and overall work with the program would be faster and more convenient (at least I think so).

**Comment by tualatin** [comment]
I know what you mean. I will see, what I can do.

**Comment by tualatin** [comment]
Hi,
I have build a prototype for that problem. Here a screenshot:

![prototype](https://user-images.githubusercontent.com/1204300/43262057-16209106-90df-11e8-9a6c-fe34da1c71ad.jpg)

Is this solution, what you expected?

**Comment by smaragdus** [comment]
@tualatin 
Yes, I imagined it this way.
Do you think that vertical layout would be easier to use? If you prefer horizontal layout please free to close this issue. I just suggested this because for me using the horizontal scroll-bar was a bit uneasy.

**Comment by tualatin** [comment]
No, I saw it like you do, that the DataGrid is too wide. I had no idea how it could look better. I think a hierarchial DataGrid could the best. In progress.

**Comment by smaragdus** [comment]
You can rely on me for testing preview builds if needed (on Windows 8),

**Comment by tualatin** [comment]
Hi,
here is a first view of the new UI:

![prototype](https://user-images.githubusercontent.com/1204300/43318654-d91c5736-91a1-11e8-9ac4-676c05225560.jpg)

**Comment by smaragdus** [comment]
@tualatin 

This looks really good!

I see you have added **SHA384**, nice. I thought of suggesting it but didn't dare since **SHA384** is not widely used.

From the screen-shot I don't see how the user switches from _lower case_ to _UPPER CASE_. Easy switch between _lower case_ and _UPPER CASE_ is preferable because in some forums _lower case_ is preferred, in others-  _UPPER CASE_. Of course this can be done via text editor or clipboard manager but for me native support for switching between _lower case_ and _UPPER CASE_ would be better.

**Comment by tualatin** [comment]
Thank you for your feedback. To configure if hashes shows in lower case or upper case, I create a settings dialog, take a look:

![settings](https://user-images.githubusercontent.com/1204300/43321591-b2e3841c-91ac-11e8-8187-96199a0575a0.jpg)

**Comment by smaragdus** [comment]
Settings dialog is a great idea!
I hope that the settings will be saved in program folder so that the program will continue to be portable.

**Comment by tualatin** [comment]
Yes, all settings save in program folder 👍 
Thank you for your constructive criticism and ideas, they helped me a lot. I think I will release a beta for you, I will post it, when I'm finish.

**Comment by tualatin** [comment]
Hi,
here is a first beta release for you. Tell me, what do you think about the new version. Thank you.

[WinHash_x64.zip](https://github.com/tualatin/winmd5checksum/files/2237382/WinHash_x64.zip)

**Comment by smaragdus** [comment]
@tualatin 

> Thank you for your constructive criticism and ideas...

In fact it was not criticism. When I come across an open source program which I find promising and think it can be improved I usually share my ideas about possible enhancements with the developer, that's it. I liked your program (tiny, portable, good set of features) so I made several suggestions.

I downloaded the Beta, I will test it now and will post about my impressions after the test.

Thanks for the Beta.

**Comment by smaragdus** [comment]
@tualatin 

I did a basic test of **WinHash** version **1.6.6783.3** and I liked this version very much- for me it is easier to work with, it requires less clicks and less scrolling and it looks better. I am also glad that last used settings are remembered after restart and 'Settings' panel also enhances program customization and usage.

A possible improvement- you may move the 'Copy hash into Clipboard' button from after the calculated hash to before the calculated hash- if ''Copy hash into Clipboard' is before the hash the user would not need to use the scroll-bar to copy the hash, copying will be a bit faster. Also, you may change ''Copy hash into Clipboard' tool-tip to just 'Copy hash'- it is shorter but this is cosmetics.

One more thing- I noticed that the program puts its icon into system tray but it does nothing. To utilize tray support you may add an option to minimize program to tray or even close to tray, perhaps two 
 new check-boxes in settings:

- Minimize to tray;
- Close to tray;

If these options are implemented the user would be able for example to minimize it to task bar and to close it to tray if 'Minimize to tray' is not checked and 'Close to tray' is checked. If the program can be minimized/closed to tray you will also have to add two tray commands, right-click on program tray icon:

- Open;
- Exit;

which commands open and exit program from tray.

I am very pleased with this Beta, I have not encountered any problems so far plus the program looks better now and working with it is easier. So for me **WinHash 1.6.6783.3** is a huge improvement, excellent work, thanks!

By the way I like the new name (**WinHash**) better.

When you release a new stable version I may write a short review about your program at a forum dedicated to portable freeware.

Regards

**Comment by tualatin** [comment]
Hello smaragdus,

thank you for your feedback. I will fix it as soon as possible. Very nice help! Thank you.

**Comment by tualatin** [comment]
I think it's final! Here you can test it again, if you want:

[WinHash_x64.zip](https://github.com/tualatin/winmd5checksum/files/2237846/WinHash_x64.zip)

* Minimize to tray
* Close to tray
* Move copy to clipboard button to left side
* Better click behavior buttons (you must not click twice, when no selection of data grid line)
* Better click behavior of checkboxes (the same here)

**Comment by smaragdus** [comment]
@tualatin 

I tested **WinHash** version **1.6.6783.21039** and this Beta is really excellent! Everything looks and works just fine! Amazing work- the program has improved a lot within a couple of days!

I think the program is ready for production release.

If/when you have time you may put some basic information about the program on [Project Main Page](https://github.com/tualatin/winmd5checksum)- for example supported hash algorithms, some of the features, a screen-shot, I think such information would be useful for new users of the program.

You may also change 'About' panel a bit putting 3 links:

- Home page - https://github.com/tualatin/winmd5checksum
- Download page - https://github.com/tualatin/winmd5checksum/releases
- Issues - https://github.com/tualatin/winmd5checksum/issues

One last thing for now, you may add a new command in 'options button'- Exit, screen:

![winhash 1 6 6783 21039 - 2018-07-28 - 002](https://user-images.githubusercontent.com/9025757/43356829-b3a61c00-9280-11e8-807f-e9f21d63c2e1.png)

In short- 'Exit' below 'About':

- About
- Separator
- Exit

'Exit' command in 'options button' would be useful especially when 'Close to tray' is enabled- the user would be able to exit program from main window.

I really like the program in its current state very much and I intend to write a review about it.

Regards

**Comment by tualatin** [comment]
smaragdus thank you very much for your feedback. I will add a exit button into the ContextMenu. You are welcome to write a review. When I have time I will add some informations about the programm to the main page.

**Comment by smaragdus** [comment]
@tualatin 

I downloaded the new version- [1.6.6783](https://github.com/tualatin/winmd5checksum/releases/tag/v.1.6.6783), tested it, liked it and as I had promised I wrote a [short review](https://www.portablefreeware.com/forums/viewtopic.php?f=4&t=24285) about **WinHash** at The Portable Freeware Forum. I hope I didn't miss something essential. I am a bit uncertain about program requirements- I supposed that **WinHash** should work on Windows 7 or higher and that it required .NET Framework 4.0 or higher. Is that true? I hope that more users would discover your nice program.

I would like to thank you again for your excellent and fast work on **WinHash**, you have improved the program very much in short time!

I may suggest several tiny enhancements in the future.

Regards

**Comment by tualatin** [comment]
@smaragdus 

Thank you very much for your review. You are right, WinHash needs Windows 7 or higher and required .NET 4.x.

Feel free to contact me, if you have any suggestions.

Regards
